src/components/AnswerEditor/index.js

```
@@ -5,50 +5,44 @@ import styled from "styled-components";
 import CustomPropTypes from "custom-prop-types";
 import DeleteButton from "components/DeleteButton";
 import TextAnswer from "components/Answers/TextAnswer";
-import CheckboxAnswer from "components/Answers/CheckboxAnswer";
-import { TEXTFIELD, CHECKBOX } from "constants/answer-types";
+import MultipleChoiceAnswer from "components/Answers/MultipleChoiceAnswer";
+import { TEXTFIELD, CHECKBOX, RADIO } from "constants/answer-types";
+import Tooltip from "components/Tooltip";
 
 export const AnswerDeleteButton = styled(DeleteButton)`
   position: absolute;
-  right: .5em;
-  top: .4em;
+  right: .8em;
+  top: .7em;
+  padding: 0;
 `;
 
-const AnswerTypeMap = {
-  [TEXTFIELD]: TextAnswer,
-  [CHECKBOX]: CheckboxAnswer
-};
-
 class AnswerEditor extends React.Component {
   handleDeleteAnswer = () => {
     this.props.onDeleteAnswer(this.props.answer.id);
   };
 
-  render() {
-    const {
-      onAddOption,
-      onUpdateOption,
-      onDeleteOption,
-      onUpdate,
-      answer
-    } = this.props;
-
-    const Answer = AnswerTypeMap[answer.type];
+  renderAnswer(answer) {
+    switch (answer.type) {
+      case TEXTFIELD:
+        return <TextAnswer {...this.props} />;
+      case CHECKBOX:
+      case RADIO:
+        return <MultipleChoiceAnswer type={answer.type} {...this.props} />;
+      default:
+        return null;
```

Do we want to throw an error here rather than return null? Returning null will just render a blank page? If we don't get a valid answer we should probably error rather trying to render nothing?

---

@samiwel In theory, this will never happen due to PropTypes for `answer` guarding against an invalid `type`: https://github.com/ONSdigital/eq-author/blob/cea60fb9068ac454040f9bb039e1acddf3012683/src/custom-prop-types/index.js#L27

---

I've actually hit an issue because of this... Added radio answer on a PR branch, switched to another branch which doesn't have radio support yet, and 💥 it blew up. `Radio` is valid in terms of it being an answer type, but it is not yet supported.

So perhaps we should throw a meaningful error, because at the moment it's some obscure react error being thrown